Source/Plugins/NetworkEvents/NetworkEvents.cpp

```
@@ -238,10 +245,72 @@ String NetworkEvents::handleSpecialMessages(const String& s)
         status += CoreServices::RecordNode::getExperimentNumber();
         return status;
     }
+    else if (cmd.compareIgnoreCase("TTL") == 0)
+    {
```

We could also return an error message rather than creating the event if acquisition is stopped. You can get that information with `CoreServices::getAcquisitionStatus()`.

---

Actually, we should test this while the queue lock is acquired below, and then override `disable` to clear the queue when acquisition is stopped. That way, we know that when we do add an event to the queue, acquisition is either active or stopping but the queue hasn't been cleared yet (since it can't be cleared until the lock is released). This would prevent stray events from waiting in the queue while it's stopped and then being added/processed when acquisition starts again.

We can also do this with the text events when they're added in the `run` function (not indicate an error in this case, but just skip adding events to the queue if acquisition is stopped). We should also switch the call to `handleSpecialMessages` and the block that adds them to the `networkMessagesQueue`, to ensure that `StopAcquisition` events consistently stop acquisition and then don't get added/processed/recorded, rather than the current situation where they might or might not get processed/recorded before acquisition stops (and if not, they get processed when it starts again). I would have done this in the PR I submitted if I had thought of it, but we can do it in this one now.

---

Implemented the first comment. Unsure about whether the second is totally needed? I think it still processes events that are in the queue even when acquisition is stopped.

---

Yeah, the second part isn't totally needed; it could be a later enhancement. If we're not adding new events while acquisition is stopped, then events not being processed in time and staying in the queue from one stop to the next start would just be an edge case.

It wouldn't process events in the queue when acquisition is stopped though, since the queue is being popped in the `process` function, which only runs when acquisition is on. Unless I'm missing something.